nemo/src/io/resource_providers.rs

```diff
@@ -43,6 +43,14 @@ impl ResourceProviders {
         Self(Rc::new(r))
     }
 
+    /// Construct default with a base path for the `FileResourceProvider`
+    pub fn default_with_base_path(base_path: Option<PathBuf>) -> Self {
```

Comment:
I'd just call this `with_base_path`, since there's, e.g., `Vec::with_capacity` and no `Vec::default_with_capacity`.

Reply:
Fine by me :)